drivers/adc/adc_mcux_adc16.c

```diff
@@ -252,19 +355,63 @@ static int mcux_adc16_init(const struct device *dev)
 #endif
 
 	ADC16_Init(base, &adc_config);
-#if defined(FSL_FEATURE_ADC16_HAS_CALIBRATION) && \
-	    FSL_FEATURE_ADC16_HAS_CALIBRATION
+#if defined(FSL_FEATURE_ADC16_HAS_CALIBRATION) &&                              \
+	FSL_FEATURE_ADC16_HAS_CALIBRATION
 	ADC16_SetHardwareAverage(base, kADC16_HardwareAverageCount32);
 	ADC16_DoAutoCalibration(base);
 #endif
 	if (config->channel_mux_b) {
 		ADC16_SetChannelMuxMode(base, kADC16_ChannelMuxB);
 	}
-	ADC16_EnableHardwareTrigger(base, false);
 
-	config->irq_config_func(dev);
+	if (IS_ENABLED(CONFIG_ADC_MCUX_ADC16_HW_TRIGGER)) {
+		ADC16_EnableHardwareTrigger(base, true);
+	} else {
+		ADC16_EnableHardwareTrigger(base, false);
+	}
+
 	data->dev = dev;
 
+	/* dma related init */
+#ifdef CONFIG_ADC_MCUX_ADC16_ENABLE_EDMA
+	/* Enable DMA. */
+	ADC16_EnableDMA(base, true);
+
+	data->adc_dma_config.dma_name = CONFIG_DMA_0_NAME;
+	data->adc_dma_config.dma_cfg.block_count = 1U;
+	data->adc_dma_config.dma_cfg.dma_slot = config->dma_slot;
+	data->adc_dma_config.dma_cfg.channel_direction = PERIPHERAL_TO_MEMORY;
+	data->adc_dma_config.dma_cfg.source_burst_length = 4U;
+	data->adc_dma_config.dma_cfg.dest_burst_length = 4U;
+	data->adc_dma_config.dma_cfg.channel_priority = 0U;
+	data->adc_dma_config.dma_cfg.dma_callback = adc_dma_callback;
+	data->adc_dma_config.dma_cfg.user_data = (void *)dev;
+
+	data->adc_dma_config.dma_cfg.source_data_size = 4U;
+	data->adc_dma_config.dma_cfg.dest_data_size = 4U;
+	data->adc_dma_config.dma_block.source_address = (uint32_t)&base->R[0];
+
+	data->dev_dma = device_get_binding(data->adc_dma_config.dma_name);
+	if (data->dev_dma == NULL) {
+		LOG_INF("dma binding fail");
```

Comment:
Should probably have a `device_is_ready` check here.

Reply:
Add check, updated